core/src/test/java/org/apache/iceberg/TestTransaction.java

```diff
@@ -771,4 +771,14 @@ public void testSimpleTransactionNotDeletingMetadataOnUnknownSate() throws IOExc
     Assert.assertTrue("Manifest file should exist", new File(manifests.get(0).path()).exists());
     Assert.assertEquals("Should have 2 files in metadata", 2, countAllMetadataFiles(tableDir));
   }
+
+  @Test
+  public void testAppendTransactionMultipleBranches(){
+    Table table = load();
+    Transaction txn = table.newTransaction();
+    txn.newAppend().toBranch("b1").appendFile(FILE_A).toBranch("b1").commit();
+    txn.newAppend().toBranch("b2").appendFile(FILE_B).toBranch("b2").commit();
+    txn.newAppend().toBranch("b2").appendFile(FILE_C).toBranch("b2").commit();
+    txn.commitTransaction();
+  }
```

Comment:
Ignore this, was just playing around here. The bulk of this change will be in the tests. I think for transactions we need to parameterize to test transactions which commit to a single branch AND we need to have tests for committing operations on different branches in a single transaction (which is where things become more interesting).

Reply:
sounds like we should first merge #5234 before we can have proper tests for this fix?